aten/src/THC/generic/THCTensorRandom.cu

```diff
@@ -490,11 +490,11 @@ THC_API void THCTensor_(clampedRandom)(THCState* state, THCTensor *self_, int64_
 #if defined(THC_REAL_IS_LONG) || defined(THC_REAL_IS_DOUBLE) || defined(THC_REAL_IS_FLOAT)
   if (range > 1ULL << 32) {
     generate_random_64<<<NUM_BLOCKS, BLOCK_SIZE, 0, THCState_getCurrentStream(state)>>>(
-        gen->state.gen_states, size, data, min_val, range);
+        gen->state.gen_states, static_cast<int32_t>(size), data, min_val, range);
```

Comment:
Why isn't a `static_cast<int>` sufficient here? The signature of the macroed function is:

```
__global__ void NAME(curandStateMtgp32 *state, int size, T *result, ARG1, ARG2)      \
```

Certainly on many platforms `int` is 32-bit, but I wouldn't rely on that.

Reply:
Yeah, that should do. I'll change that.